### PR TITLE
samples: matter: Added config to inform about experimental support

### PR DESCRIPTION
**Pull request by kkasperczyk-no**
* Added Matter Kconfig to subsys
* Added CHIP config selecting EXPERIMENTAL support for Matter
that will be merged with existing CHIP upstream config.

Signed-off-by: Kamil Kasperczyk <kamil.kasperczyk@nordicsemi.no>

### CODE REVIEW

File: subsys/matter/Kconfig [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Copyright (c) 2021 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+# Support for Matter protocol in NCS is experimental
+config CHIP
+	bool "Matter (formerly Connected Home over IP) protocol stack"
```
Comment (Damian-Nordic) [inline subsys/matter/Kconfig:8]: I think you don't even need to repeat the title and help. It will be taken from the upstream, right?

Comment (kkasperczyk-no) [reply subsys/matter/Kconfig:8]: Yes, I just repeated it for the code readability purpose, but I can either remove it.

Comment (greg-fer) [reply subsys/matter/Kconfig:8]: I'd drop "formerly CHIP" here. No need to mention it at every corner in my opinion. @LuDuda ?

File: doc/nrf/releases/release-notes-changelog.rst [hunk 2]
```diff
@@ -51,6 +51,11 @@ Bluetooth
 
 * |no_changes_yet_note|
 
+Matter
+------
+
+* Added selecting config informing that Matter support is experimental.
```
Comment (greg-fer) [inline doc/nrf/releases/release-notes-changelog.rst:57]:
```suggestion
* Added ``EXPERIMENTAL`` select in Kconfig that informs that Matter support is experimental.
```

**Review by greg-fer — APPROVED** [review]

**Review by Damian-Nordic — APPROVED** [review]

**Review by LuDuda — APPROVED** [review]

**Pull request merged**